synth/src/lib.rs

```diff
@@ -1,7 +1,7 @@
 #![feature(format_args_capture, async_closure, map_first_last, box_patterns)]
 #![feature(error_iter)]
 #![allow(type_alias_bounds)]
-#![deny(warnings)]
+// #![deny(warnings)]
```

Comment:
What warnings are we allowing now?

Not that I have any problem with this, as I personally consider denying warnings an [antipattern](https://rust-unofficial.github.io/patterns/anti_patterns/deny-warnings.html), since that is better specified in CI, but we should at least be clear what this entails.

Reply:
Yeah funnily enough I was looking at the rust-patterns GH repo and this turned up. I'll create an issue for this.